tools/validate_spec.py

```diff
@@ -0,0 +1,348 @@
+#!/usr/bin/env python3
```

Comment:
Running Python optimized (`#!/usr/bin/env -S python3 -O`) would improve the performance, as assertions and contracts are not executed then.

Reply:
Added in 2aafab5